test/unit/security/test_vault.py

```diff
@@ -0,0 +1,61 @@
+import os
+import unittest
+
+from galaxy.app_unittest_utils.galaxy_mock import MockApp, MockAppConfig
```

Comment:
So ``galaxy.security`` belongs to galaxy-data which has a restricted set of packages it depends on and definitely doesn't depend on ``galaxy-app`` - the bulk of the application. So this line will never work in the decomposed version of Galaxy.

Some possible options include -

1. Just refactoring the whole thing into ``galaxy.vault``.
2. putting the interfaces and the database implementation in galaxy.security and sticking other implementations else where in ``galaxy-app`` - like ``galaxy.vault``.
3. Refactor the unit tests so that stuff that requires app are tested in galaxy-data and then more integration-y unit tests are placed in galaxy-app. I did something like this with https://github.com/galaxyproject/galaxy/pull/12665/files for instance. Where you can see I refactored unit tests so galaxy-files could be properly packaged and then added a new unit test for how it interacts with sniffers (in galaxy-data).

I think 1 is the quickest, easiest thing to do but I suspect something like a combination of 2 and 3 would let us have the strongest typing and clearest breakdown of dependencies.

This code seems well tested and well thought through - if you don't care about how the package structure and dependency structure breaks down I can take a crack at refactoring it once it is ready.

Reply:
> ... The only reason for needing the MockApp was to get at an sa_session for the database vault backend, which I suppose should be doable since the models are a part of the same package. Is there an easy way to get at that, without needing to initialize the mappings manually?

Maybe something [like this](https://github.com/galaxyproject/galaxy/blob/dev/test/unit/data/model/mapping/conftest.py)? That's an engine fixture (connection to whatever test db you need) + a session fixture (that gives you a managed scoped session) + an init_model fixture that initialized the database (passed to the session fixture, module-scoped - so executed only once per module). Here's an example of the init_model in use: https://github.com/galaxyproject/galaxy/blob/dev/test/unit/data/model/mapping/test_model_mapping.py#L6869 (`model` is imported here: https://github.com/galaxyproject/galaxy/blob/dev/test/unit/data/model/mapping/test_model_mapping.py#L69).

Reply:
There is a smaller version of app that does setup the sa_session in https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/model/unittest_utils/data_app.py#L64.

Reply:
@ic4f I added the database test.

@jmchilton That smaller version worked well. I think the PR is ready to review now. Thanks for offering to change the package structure - happy to take you up on it.